Project/Tsofen/compare_source_target.py

```diff
@@ -1,63 +1,65 @@
-import json
-import os
-from enums import Status
-import re
-import logging
-logging.basicConfig(level=logging.DEBUG, format='%(asctime)s - %(levelname)s - %(message)s')
 
+from Job import *
 
-class CompareSourceTarget:
 
-    def __init__(self):
-        self.file_name = None
-        self.__data=None
+class CompareSourceTarget(Job):
 
-    def run(self, file_name: str = "json_file.json"):
+    def __init__(self):
+        """
+        constructor
+        """
+        super().__init__()
+        self.__version_file_name = None
+    def run(self, email_addresses, version_file_name: str = "json_file.json",email_file_name: str = "YAML.json"):
         """
         this function is the entering point of the job.
         """
-        self.file_name = file_name
-        self.__prerequisite()
+        self._email_addresses=email_addresses
+        self.__version_file_name = version_file_name
+        self._email_file_name=email_file_name
+        super()._prerequisite()
+        self._prerequisite()
```

Comment:
```suggestion
        super()._prerequisite()
        self._prerequisite()
```
please move the call to prerequisite and super prerequisite to be in the first lines of "run" function